internal/git/gitignore.go

```diff
@@ -81,10 +61,16 @@ func (r repoIgnoreTester) Matches(f string, isDir bool) (bool, error) {
 	return r.gitIgnoreTester.Matches(f, isDir)
 }
 
-func NewRepoIgnoreTester(ctx context.Context, repoRoot string) (model.PathMatcher, error) {
-	g, err := NewGitIgnoreTester(ctx, repoRoot)
+func NewRepoIgnoreTester(ctx context.Context, repoRoot string, gitignoreContents string) (*repoIgnoreTester, error) {
+	g, err := NewGitIgnoreTesterFromContents(ctx, repoRoot, gitignoreContents)
 	if err != nil {
 		return nil, err
 	}
 	return &repoIgnoreTester{repoRoot, g}, nil
 }
+
+type nothingMatcher struct{}
```

Comment:
there should already be something like this in the main model package?

Reply:
Yes but I can't include the `model` package because then I have a circular dependency. :(

I could try moving it out in to another package though.

Reply:
After talking with @nicks I'm going to reorganize the model package so that it doesn't depend on the gitignore or dockerignore packages in a separate PR.